src/react/01 - controlled components/fullyuncontrolled.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+class FullyUnControlled extends React.Component {
+	static propTypes = {
+		handleBlur: PropTypes.func.isRequired,
+		type: PropTypes.string,
+	}
+
+	static defaultProps = {
+		type: 'text',
+	}
+
+	handleBlur = () => {
+		this.props.handleBlur(this.input.value)
+	}
+
+	render() {
+		const {type} = this.props
+
+		return (
+			<React.Fragment>
+				<label>Fully Un-Controlled:</label>
+				<input
+					className='awesome-input'
+					type={type}
+					onBlur={this.handleBlur}
+					placeholder='Name...'
+					ref={(el) => {
```

Comment:
This is the old way of creating Ref's. Have a look into the 'new' React.createRef();

https://reactjs.org/docs/refs-and-the-dom.html

Reply:
thanks. updated it.